@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1033,28 +1042,38 @@ export class TextElementsRenderer {
     private prepareTextElementGroup(
         textElementGroup: TextElementGroup,
         tile: Tile,
-        worldMaxX: number,
+        projection: Projection,
         maxViewDistance?: number
     ) {
         if (textElementGroup.elements.length === 0) {
             return;
         }
 
-        const worldOffsetX = worldMaxX * tile.offset;
+        const worldOffsetX = projection.worldExtent(0, 0).max.x * tile.offset;
 
         const textElementSelection: TextElementFilter = (
             textElement: TextElement,
             lastFrameNumber?: number
         ): number | undefined => {
-            const { result, viewDistance } = checkReadyForPlacement(
+            let { result, viewDistance } = checkReadyForPlacement(
                 textElement,
                 tile,
                 worldOffsetX,
-                this.m_mapView,
-                this.m_textElementStateCache,
-                maxViewDistance,
-                lastFrameNumber
+                this.m_viewState,
+                this.m_viewCamera,
+                this.m_poiManager,
+                projection.type,
+                maxViewDistance
             );
+
+            if (
+                result === PrePlacementResult.Ok &&
+                !this.m_textElementStateCache.deduplicateElement(textElement, lastFrameNumber)
```

Comment:
just moved this call to deduplicate elements, before it was within checkReadyForPlacement.